### PR TITLE
doc: update tag file with respect to structure changes

### PR DESCRIPTION
**Pull request by divipillai**
* Update the tag file with respect to the
 top level structural changes
* Added tags for nRF9131.
* Added tags for nRF9151

### CODE REVIEW

**Review by umapraseeda — CHANGES_REQUESTED** [review]

File: doc/tags.yml [hunk 1]
```diff
@@ -28,154 +29,173 @@ mapping_topics:
...
+  - nrf/app_dev/device_guides/nrf91/nrf91_board_controllers.html: ["nrf9160", "nrf9161",
+                                                                   "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_cloud_certificate.html: ["nrf9160", "nrf9161",
+                                                                   "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/thingy91_connecting.html: ["thingy91", "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_dk_updating_fw_programmer.html: ["nrf9160", "nrf9161",
+                                                                           "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/thingy91_updating_fw_programmer.html: ["nrf9160", "nrf9161",
+                                                                           "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_building.html: ["nrf9160", "thingy91", "nrf9161",
```
[the remark this answers is not in the document]
Comment (divipillai) [reply doc/tags.yml:121]: yeah, its applicable as the page contain instructions on building for all the nRF91 Series devices.

File: doc/tags.yml [hunk 2]
```diff
@@ -28,154 +29,173 @@ mapping_topics:
...
+  - nrf/app_dev/device_guides/nrf91/nrf91_programming.html: ["nrf9160", "thingy91", "nrf9161",
+                                                             "development-kits",
+                                                             "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_testing_at_client.html: ["nrf9160", "nrf9161",
+                                                                   "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_snippet.html: ["nrf9160", "thingy91", "nrf9161",
+                                                         "development-kits",
+                                                         "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf9160_external_flash.html: ["nrf9160", "development-kits"]
+  - nrf/app_dev/device_guides/nrf70/index.html: ["nrf70-series", "nrf7002", "nrf7001", "nrf7000",
```
[the remark this answers is not in the document]
Comment (divipillai) [reply doc/tags.yml]: Could see similar cases in other places as well. Hence, fixed whenever applicable.

File: doc/tags.yml [hunk 3]
```diff
@@ -28,159 +29,151 @@ mapping_topics:
...
+  - nrf/app_dev/device_guides/nrf91/index.html: ["nrf9160", "thingy91", "nrf9161", "nrf9151",
+                                                 "nrf9131", "development-kits",
+                                                 "prototyping-platforms", "dect-nr+"]
+  - nrf/app_dev/device_guides/nrf91/*.html: ["nrf91-series", "lte"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_features.html: ["nrf9160", "thingy91", "nrf9151",
+                                                          "nrf9161", "development-kits", "dect-nr+",
+                                                          "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_board_controllers.html: ["nrf9160", "nrf9161", "nrf9151",
+                                                                   "nrf9161", "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_cloud_certificate.html: ["nrf9160", "nrf9161", "nrf9151",
```
Comment (richabp) [inline doc/tags.yml:93]: Should "nrf-cloud" be added here?

File: doc/tags.yml [hunk 4]
```diff
@@ -287,27 +274,15 @@ mapping_topics:
...
                                               "nrf52832", "nrf52820", "nrf52811", "nrf52810",
                                               "nrf52805", "development-kits"]
-  - nrf/libraries/bluetooth_services/mesh.html: ["ble", "blemesh", "nrf52-series", "nrf52840",
-                                                 "nrf52833", "nrf52832", "nrf52820", "nrf52811",
-                                                 "nrf52810", "nrf52805", "development-kits"]
-  - nrf/libraries/bluetooth_services/mesh/*.html: ["ble", "blemesh", "nrf52-series", "nrf52840",
-                                                   "nrf52833", "nrf52832", "nrf52820", "nrf52811",
-                                                   "nrf52810", "nrf52805", "development-kits"]
-  - nrf/libraries/gazell/index.html: ["gazell"]
+  - nrf/libraries/bluetooth_services/mesh.html: ["blemesh"]
```
Comment (richabp) [inline doc/tags.yml:283]: Is there a specific reason for removing other tags from the mesh folder?

Comment (divipillai) [reply doc/tags.yml:283]: The reason to remove is, we already have these tags in "nrf/libraries/bluetooth_services/*.html". So, those will be applied.

File: doc/tags.yml [hunk 5]
```diff
@@ -28,159 +29,151 @@ mapping_topics:
...
-  - nrf/device_guides/fem/*.html: ["fem", "nrf21540", "evaluation-kits", "development-kits"]
-  - nrf/device_guides/wifi_coex.html: ["kconfig", "ble", "nrf70-series", "nrf7002", "nrf7001",
-                                       "nrf7000", "wifi"]
+                                                        "nrf9151", "nrf9131", "lte"]
   - nrf/security/security.html: ["kconfig"]
   - nrf/security/ap_protect.html: ["kconfig"]
   - nrf/protocols/*.html: ["protocols"]
   - nrf/protocols.html: ["sidewalk", "ble", "blemesh", "esb", "gazell", "matter", "multiprotocol",
-                         "nfc", "thread", "zigbee", "wifi", "dect-nr+"]
+                         "nfc", "thread", "zigbee", "wifi", "dect-nr+", "protocols"]
```
Comment (umapraseeda) [inline doc/tags.yml:176]: Protocols is already added to this page through the line above.
```suggestion
                         "nfc", "thread", "zigbee", "wifi", "dect-nr+"]
```

File: doc/tags.yml [hunk 6]
```diff
@@ -28,159 +29,151 @@ mapping_topics:
...
-  - nrf/config_and_build/bootloaders/bootloader_config.html: ["kconfig"]
-  - nrf/config_and_build/dfu/*.html: ["applications"]
+  - nrf/app_dev/bootloaders_dfu/*.html: ["applications"]
+  - nrf/app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_adding.html: ["kconfig"]
+  - nrf/app_dev/bootloaders_dfu/mcuboot_nsib/bootloader_config.html: ["kconfig"]
+  - nrf/app_dev/device_guides/nrf91/index.html: ["nrf9160", "thingy91", "nrf9161", "nrf9151",
+                                                 "nrf9131", "development-kits",
+                                                 "prototyping-platforms", "dect-nr+"]
+  - nrf/app_dev/device_guides/nrf91/*.html: ["nrf91-series", "lte"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_features.html: ["nrf9160", "thingy91", "nrf9151",
```
Comment (umapraseeda) [inline doc/tags.yml:88]: missing 9131

File: doc/tags.yml [hunk 7]
```diff
@@ -28,159 +29,151 @@ mapping_topics:
...
+                                                                           "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_building.html: ["nrf9160", "thingy91", "nrf9161",
+                                                          "nrf9151", "development-kits",
+                                                          "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_programming.html: ["nrf9160", "thingy91", "nrf9161",
+                                                             "nrf9151", "development-kits",
+                                                             "prototyping-platforms"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_testing_at_client.html: ["nrf9160", "nrf9161",
+                                                                   "nrf9151", "development-kits"]
+  - nrf/app_dev/device_guides/nrf91/nrf91_snippet.html: ["nrf9160", "thingy91", "nrf9161",
```
Comment (umapraseeda) [inline doc/tags.yml:109]: missing 9131

File: doc/tags.yml (hunk 5 above)
Comment (divipillai) [reply doc/tags.yml:176]: The above line is for the files under the protocols folder. This page is outside the protocol's folder.

File: doc/tags.yml [hunk 8]
```diff
@@ -237,8 +225,7 @@ mapping_topics:
   - nrf/applications/asset_tracker_v2/README.html: ["lte"]
```
[the remark this answers is not in the document]
Comment (divipillai) [reply doc/tags.yml:225]: I think we should keep this as the above path is "nrf/applications/asset_tracker_v2/doc/*.html", when the tag is applied. If we remove this page, readme.html won't have the tag.

File: doc/tags.yml [hunk 9]
```diff
@@ -325,7 +294,6 @@ mapping_topics:
                                                      "nrf7000"]
```
Comment (umapraseeda) [inline doc/tags.yml:294]: We can also add tags for https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/networking/softap_wifi_provision.html and https://docs.nordicsemi.com/bundle/ncs-latest/page/nrf/libraries/networking/wifi_ready.html

File: doc/tags.yml [hunk 10]
```diff
@@ -335,7 +303,6 @@ mapping_topics:
   - nrf/libraries/others/esb.html: ["esb"]
   - nrf/libraries/others/fem_al.html: ["fem", "nrf21540"]
   - nrf/libraries/others/st25r3911b_nfc.html: ["nfc"]
-  - nrf/libraries/zigbee/index.html: ["zigbee"]
   - nrf/libraries/zigbee/*.html: ["zigbee"]
   - nrf/external_comp/nrf_cloud.html: ["nrf-cloud"]
   - nrf/external_comp/avsystem.html: ["lte"]
```
Comment (umapraseeda) [inline doc/tags.yml:308]: For nrf/releases_and_maturity/software_maturity.html: ["sidewalk", "ble", "blemesh", "matter",
                                                       "thread", "zigbee", "wifi", "lte"] below, dect nr+ could be added.

File: doc/tags.yml [hunk 11]
```diff
@@ -347,30 +314,19 @@ mapping_topics:
...
   - nrfxlib/*.html: ["libraries"]
-  - nrfxlib/gzll/README.html: ["gazell"]
   - nrfxlib/gzll/*.html: ["gazell"]
-  - nrfxlib/lc3/README.html: ["nrf5340", "ble"]
   - nrfxlib/lc3/*.html: ["nrf5340", "ble"]
-  - nrfxlib/nrf_modem/README.html: ["nrf91-series", "nrf9160", "nrf9161", "lte", "dect-nr+"]
-  - nrfxlib/nrf_modem/*.html: ["nrf91-series", "nrf9160", "nrf9161", "lte", "dect-nr+"]
+  - nrfxlib/nrf_modem/*.html: ["nrf91-series", "nrf9160", "nrf9161", "nrf9151", "lte", "dect-nr+"]
   - nrfxlib/nrf_modem/doc/at_interface.html: ["at-commands"]
   - nrfxlib/nrf_802154/README.html: ["nrf52-series", "nrf52840", "nrf52833", "nrf52832", "nrf52820",
```
Comment (umapraseeda) [inline doc/tags.yml:321]: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrfxlib/nrf_modem/doc/dectphy.html could benefit from a dect nr+ tag.

File: doc/tags.yml [hunk 12]
```diff
@@ -347,30 +314,19 @@ mapping_topics:
...
   - nrfxlib/nrf_fuel_gauge/*.html: ["pmic", "npm1100", "npm1300", "npm6001", "evaluation-kits"]
-  - nrfxlib/nfc/README.html: ["nrf52-series", "nrf53-series", "nfc"]
   - nrfxlib/nfc/*.html: ["nrf52-series", "nrf53-series", "nfc"]
-  - nrfxlib/openthread/README.html: ["thread"]
   - nrfxlib/openthread/*.html: ["thread"]
-  - nrfxlib/nrf_dm/README.html: ["nrf52-series", "nrf53-series"]
   - nrfxlib/nrf_dm/*.html: ["nrf52-series", "nrf53-series"]
-  - nrfxlib/nrf_wifi/README.html: ["wifi", "nrf70-series", "nrf7002", "nrf7001", "nrf7000",
-                                   "drivers"]
   - nrfxlib/nrf_wifi/doc/*.html: ["wifi", "nrf70-series", "nrf7002", "nrf7001", "nrf7000",
```
Comment (umapraseeda) [inline doc/tags.yml:328]:
```suggestion
  - nrfxlib/nrf_wifi/*.html: ["wifi", "nrf70-series", "nrf7002", "nrf7001", "nrf7000",
```

File: doc/_zoomin/ncs.tags.yml [hunk 13]
```diff
@@ -247,14 +228,12 @@ mapping_topics:
                        "nrf52-series", "nrf7002", "nrf7001", "nrf7000", "nrf9160", "thingy91",
                        "nrf9161", "nrf9151", "nrf54h20", "nrf54l15", "nrf5340", "thingy53",
                        "nrf52840", "nrf52833", "nrf52832", "nrf52820", "nrf52811", "nrf52810",
-                       "nrf52805", "nrf21540", "npm1100", "npm1300", "npm6001", "development-kits",
-                       "prototyping-platforms", "kconfig"]
+                       "nrf52805", "nrf21540", "npm1100", "npm1300", "npm6001",
+                       "development-kits", "prototyping-platforms", "kconfig"]
```
Comment (umapraseeda) [inline doc/_zoomin/ncs.tags.yml:232]: development-kits fit in the earlier line.

**Review by umapraseeda — APPROVED** [review]

**Review by richabp — APPROVED** [review]

**Pull request merged**